bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoAbstractHandler.java

```diff
@@ -67,7 +71,10 @@
     protected static final int MAX_QUEUE = 5;
     protected static final Gson GSON = new GsonBuilder().create();
 
+    protected ScheduledExecutorService miIoScheduler = ThreadPoolManager.getScheduledPool(BINDING_ID);
```

Comment:
If since you need it to be non-null, but are going to override it anyway during initialization, then I see no problem with initially setting it to the BaseThingHandler scheduler.
```suggestion
    protected ScheduledExecutorService miIoScheduler = scheduler;
```

Reply:
yes, indeed that was the intention, will change it to your suggestion.